src/palace/manager/integration/patron_auth/saml/metadata/model.py

```diff
@@ -1045,7 +1045,7 @@ class SAMLSubject:
 
     def __init__(
         self,
-        idp: str,
+        idp: str | None,
```

Comment:
An existing tests checked the we got the correct response with `idp` is `None`. 

As an incidental part of the refactoring in this PR that test got type hints, which started raising a type error because we were passing `None` to this parameter. Based on the test, the type hint on this class was incorrect, so I changed it as part of this PR.